### PR TITLE
Updates for WinAppSDK 1.2

### PR DESCRIPTION
**Pull request by bpulliam**
## Description

This adds to the work Mano did and ensures that all samples meet the following:
- Pointing to WinAppSDK 1.2 (currently preview 2, but will be updated to 1.2 stable upon release)
- Minimum versions:
    NET 6
    CsWinRT 2.0.0
    CppWinRT 2.0.220929.3
    Windows Implementation Library 1.0.220914.1
    Windows SD Build Tools 10.0.22621.755
    

Note: The following samples have pre-existing issues and still need to be updated:
- D3D9ExSample
- DynamicRefreshRateSample
- DeploymentManagerSample

## Target Release

WinAppSDK 1.2 stable.

## Checklist

- [ ] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [ ] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [ ] Samples set the minimum supported OS version to Windows 10 version 1809.
- [ ] Samples build clean with no warnings or errors.
- [ ] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).
- [ ] If I am onboarding a new feature, then I must have correctly setup a new CI pipeline for my feature with the correct triggers and path filters laid out in the "Onboarding Samples CI Pipeline for new feature" section in samples-guidelines.md.
- [ ] I have commented on my PR `/azp run SamplesCI-<FeatureName>` to have the CI build run on my branch for each of my FeatureName my PR is modifying. This must be done on the latest commit on the PR before merging to ensure the build is up to date and accurate. Warning: the PR will not block automatically if this is not run due to '/azp run' limitation on triggering more than 10 pipelines.

### CODE REVIEW

**Review by pratikone — APPROVED** [review]
:shipit:

**Review by pratikone — APPROVED** [review]

File: Samples/AppLifecycle/Restart/cpp-console-unpackaged/cpp-console-unpackaged.vcxproj [hunk 1]
```diff
@@ -207,21 +207,38 @@
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
+<<<<<<< HEAD
```
Comment (manodasanW) [inline Samples/AppLifecycle/Restart/cpp-console-unpackaged/cpp-console-unpackaged.vcxproj:210]: Conflict marker here and below.

File: Samples/DeploymentManager/cs-winui/DeploymentManagerSample.csproj [hunk 2]
```diff
@@ -10,13 +10,13 @@
     <UseWinUI>true</UseWinUI>
     <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
     <PublishProfile>Properties\PublishProfiles\win10-$(Platform).pubxml</PublishProfile>
-	<WindowsSdkPackageVersion>10.0.19041.24</WindowsSdkPackageVersion>
+	<WindowsSdkPackageVersion>10.0.19041.27</WindowsSdkPackageVersion>
```
Comment (manodasanW) [inline Samples/DeploymentManager/cs-winui/DeploymentManagerSample.csproj:13]: We can probably drop this property, 27 should be the version in the .NET 6 SDK used by the pipeline.

Comment (bpulliam) [reply Samples/DeploymentManager/cs-winui/DeploymentManagerSample.csproj:13]: I'm going to do this separately.

**Pull request merged**